src/osx/cocoa/dataview.mm

```diff
@@ -2508,6 +2508,19 @@ void UpdateWithRow(int row)
             byExtendingSelection:GetDataViewCtrl()->HasFlag(wxDV_MULTIPLE) ? YES : NO];
 }
 
+void wxCocoaDataViewControl::Select(const wxDataViewItemArray& items)
+{
+    NSMutableIndexSet *selection = [[NSMutableIndexSet alloc] init];
+
+    for (wxDataViewItemArray::const_iterator i = items.begin(); i != items.end(); ++i)
```

Comment:
FWIW we use C++11 in macOS now (as it's always available on all the still supported platforms), so this could be written using range for.